src/raft.rs

```diff
@@ -516,42 +516,47 @@ impl<T: Storage> Raft<T> {
 
         m.set_msg_type(MessageType::MsgSnapshot);
         let snapshot_r = self.raft_log.snapshot(pr.pending_request_snapshot);
-        if let Err(e) = snapshot_r {
-            if e == Error::Store(StorageError::SnapshotTemporarilyUnavailable) {
+        match snapshot_r {
```

Comment:
In my option, using match here makes the code ugly.

Reply:
> In my option, using match here makes the code ugly.

not bad for me, Does anyone else have any other opinions? It's necessary to rollback the code？

Reply:
I think both are OK - using `if let` and an early return feels a bit more clear, but a bit more C/Go/Swift style, whereas `match` seems more like idiomatic Rust, but a bit harder to read.